mitodl/edx-platform
# Reimplemented legacy remote gradebook merge/overload functionality

### ⚠️  REBASED ON #47 ⚠️

#### Manual testing steps (copied from #20)

- Make sure you have staff access for your test course
- Set the `ENABLE_INSTRUCTOR_REMOTE_GRADEBOOK_CONTROLS` feature to `true` (in `lms.env.json`)
- Create a `REMOTE_GRADEBOOK` setting in `lms.env.json`. This should be a dict with 2 values: 
    ```
    "REMOTE_GRADEBOOK": {
        "URL": "https://lmodproxyqa.odl.mit.edu/edx_grades",
        "DEFAULT_NAME": "STELLAR:/project/mitxdemosite"
    }
    ```

- Create `REMOTE_GRADEBOOK_USER` and `REMOTE_GRADEBOOK_PASSWORD` settings in `lms.auth.json`. Ask me or Peter for these values
- Navigate to the `Remote Gradebook` tab for some course (eg: `http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-remote_gradebook`)
- Click the "Merge"/"Overload enrollment list using remote gradebook" buttons and make sure the results are as expected. "Merge" is intended to take all users for a specific section in the remote gradebook and enroll them in the given course in edX. "Overload" does the same thing, but also unenrolls all currently enrolled non-staff users for the given course. If you have such users in your local environment, you should be prompted to confirm that you want to unenroll those users
